google/cloud/internal/async_rpc_details.h

```diff
@@ -63,6 +63,17 @@ class AsyncUnaryRpcFuture : public AsyncGrpcOperation {
              Request const& request, grpc::CompletionQueue* cq, void* tag) {
     promise_ = promise<StatusOr<Response>>([context] { context->TryCancel(); });
     auto rpc = async_call(context.get(), request, cq);
+    void* intial_metadata_tag;
+    bool inital_metadata_ok = false;
+    auto deadline = std::chrono::system_clock::now() + std::chrono::seconds(30);
+    cq->AsyncNext(&intial_metadata_tag, &inital_metadata_ok, deadline);
```

Comment:
This doesn't look right. Why are we doing an `AsyncNext` before the operation?

Reply:
I thought we needed to get the tag before doing the operation intial_metadata_tag